lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -549,14 +549,15 @@ def commit(method, url, parameters = nil, options = {})
         add_expand_parameters(parameters, options) if parameters
         response = api_request(method, url, parameters, options)
 
-        success = !response.key?("error")
+        success = !response.key?("error") && response["status"] != "failed"
+        error_message = response.fetch("error", {"message" => "No details provided"})["message"]
```

Comment:
"No error details" might be a little more explicit than "No details provided".

Reply:
Good call; will change.